beacon_node/network/src/network_beacon_processor/rpc_methods.rs

```diff
@@ -93,41 +93,67 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
             // current slot. This could be because they are using a different genesis time, or that
             // their or our system's clock is incorrect.
             Some("Different system clocks or genesis time".to_string())
-        } else if (remote.finalized_epoch == local.finalized_epoch
-            && remote.finalized_root == local.finalized_root)
-            || remote.finalized_root.is_zero()
-            || local.finalized_root.is_zero()
-            || remote.finalized_epoch > local.finalized_epoch
-        {
-            // Fast path. Remote finalized checkpoint is either identical, or genesis, or we are at
-            // genesis, or they are ahead. In all cases, we should allow this peer to connect to us
-            // so we can sync from them.
-            None
         } else {
-            // Remote finalized epoch is less than ours.
-            let remote_finalized_slot = start_slot(remote.finalized_epoch);
-            if remote_finalized_slot < self.chain.store.get_oldest_block_slot() {
-                // Peer's finalized checkpoint is older than anything in our DB. We are unlikely
-                // to be able to help them sync.
-                Some("Old finality out of range".to_string())
-            } else if remote_finalized_slot < self.chain.store.get_split_slot() {
-                // Peer's finalized slot is in range for a quick block root check in our freezer DB.
-                // If that block root check fails, reject them as they're on a different finalized
-                // chain.
-                if self
-                    .chain
-                    .block_root_at_slot(remote_finalized_slot, WhenSlotSkipped::Prev)
```

Comment:
If this check is indeed cheap, I would prefer to keep it as it would cause make it harder for peers to sync in a chaotic network if we remove this.
I do not have a sense of how cheap this is though.

Reply:
It's very cheap, reading a single block root from the freezer is fast (<3ms)